### DISCUSSION TITLE
Starting playback on a hardware synth + syncing tempo

### DISCUSSION THREAD

**Discussion by phoria-mark-a** [melanchall/drywetmidi#265] [Q&A] [answered]
Hi there,

First of all great work on DryWetMidi, having a lot of fun diving in.

Wondering if it's possible to control an external synth (e.g. Elektron Syntakt, Eurorack via Intellijel MIDI1U) with DWM?

Using Start and Stop MidiEvents I can get the Syntakt to go into play mode, and then stop (though the stop command causes it to pause which is odd), however the problem is that it doesn't actually progress the local sequence, it's just stuck on step 1.

I've also tried manipulating the BPM on the device using DryWetMidi, without any luck so far. I've tried sending TimingClockEvents, but it's clearly the wrong way to go about it, as it jumps up to its maximum BPM of 300.

Ideally I'd love to be able to send clock and run/transport to hardware synths, so that they can play their own programmed sequences locally, and then use DWM to manipulate additional parameters via MIDI.

I'm using the Unity package if that affects anything, but I think I must just be missing something pretty obvious.

Any help would be much appreciated.

Kindest,
Mark

**Comment by melanchall** [comment, marked as answer]
Hi Mark,

First of all, thank you for using the library.

Well, as for controlling an external synth, you need to refer to the device manual. DWM allows you to send MIDI events, but what events to send exactly to your device and what order of them to use is completely up to the device. So sorry, but unfortunately I can't say how to control the device with MIDI events.

Thanks,
Max

**Comment by phoria-mark-a** [comment]
Hi Max,

Thanks so much for the response.

I guess a more specific question would be - what is the best way to synchronise multiple external MIDI devices?

Or failing that, is it possible to receive the time information of an external synth to use in Unity? If I use a midi device as the master clock, send TimingClock event to Unity, which passes the event to a slave device, it all syncs up, however Unity is unaware as to the current clock information. I'd love to be able to have that context in Unity to trigger things at certain times.

I'll keep diving in, just feels like I need a course correction or something.

Kindest,
Mark

**Reply by melanchall** [reply]
Hi Mark,

If I understand you correctly, you need the following steps:

1. receive timing clock events from synths;
2. start `Stopwatch` in your Unity game;
3. when a clock "pulse" received, just get `.Elapsed` of the `Stopwatch`.

Thanks,
Max
